src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,54 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public class Block {
+    public static final String EXAMPLE = "123";
+    public static final String MESSAGE_BLOCK_CONSTRAINTS = "Block has to be in numerals";
+    public static final String BLOCK_VALIDATION_REGEX = "\\d{1,3}.";
+
+    public String value;
+    protected boolean isPrivate;
+
+    /**
+     * Validates given Block.
+     *
+     * @throws IllegalValueException if given Block string is invalid.
+     */
+    public Block(String Block, boolean isPrivate) throws IllegalValueException {
+        String trimmedBlock = Block.trim();
+        this.isPrivate = isPrivate;
+        if (!isValidBlock(trimmedBlock)) {
+            throw new IllegalValueException(MESSAGE_BLOCK_CONSTRAINTS);
+        }
+        this.value = trimmedBlock;
```

Comment:
[Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! Avoid unnecessary use of `this` with fields: Use the `this` keyword only when a field is shadowed by a method or constructor parameter.